### PR TITLE
atkmm, cairomm, glibmm, pangomm: bump and rename

### PR DESCRIPTION
**Pull request by amaxine**
###### Motivation for this change
Individual commits link to changelogs (when available). For the sake of consistency, I renamed the packages to include the ABI version (following how GNOME refers to them), largely in anticipation of this becoming more annoying/confusing otherwise later.

All version bumps are in line with what GNOME 40.4 expects so I'm not concerned about doing this ahead of 41.

I think I'm kinda messing up on the update script for `atkmm_2_36`... I'll look at fixing that later, unless someone knows off the top of their head how to do it right :)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by maxeaubrey** [comment]
I'm currently running nixpkgs-review against this PR, but that's going to take a while...

**Comment by maxeaubrey** [comment]
Result of `nixpkgs-review pr 136893` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>gnome.gnome-documents</li>
    <li>jucipp</li>
    <li>synfigstudio</li>
    <li>webbrowser</li>
    <li>xfce.xfce4-hardware-monitor-plugin</li>
  </ul>
</details>
<details>
  <summary>155 packages failed to build:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>adriconf</li>
    <li>ahoviewer</li>
    <li>aj-snapshot</li>
    <li>arc-theme</li>
    <li>ardour_5</li>
    <li>arx-libertatis</li>
    <li>asciidocFull (asciidoc-full)</li>
    <li>asciidoc-full-with-plugins</li>
    <li>astroid</li>
    <li>audiality2</li>
    <li>bibata-cursors</li>
    <li>bibata-cursors-translucent</li>
    <li>bibata-extra-cursors</li>
    <li>blueberry</li>
    <li>bombono</li>
    <li>btrbk</li>
    <li>capitaine-cursors</li>
    <li>carla</li>
    <li>cherrytree</li>
    <li>chrome-gnome-shell</li>
    <li>cinnamon.cinnamon-screensaver</li>
    <li>cinnamon.iso-flags-png-320x420</li>
    <li>cinnamon.iso-flags-svg</li>
    <li>clevis</li>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>dablin</li>
    <li>dblatexFull</li>
    <li>disorderfs</li>
    <li>easyeffects (pulseeffects-pw)</li>
    <li>emojione</li>
    <li>faust2jack</li>
    <li>faust2jaqt</li>
    <li>faustPhysicalModeling</li>
    <li>faustStk</li>
    <li>ffado (ffadoFull)</li>
    <li>fim</li>
    <li>flowblade</li>
    <li>gImageReader</li>
    <li>giada</li>
    <li>gigedit</li>
    <li>glom</li>
    <li>gnome.gnome-session</li>
    <li>gnome.gnome-shell</li>
    <li>gnome.gnome-system-monitor</li>
    <li>gnome.gnome-terminal</li>
    <li>gnome.gnome-tweak-tool</li>
    <li>gnome.gtkmm</li>
    <li>gnome.gtksourceviewmm</li>
    <li>gnome.nemiver</li>
    <li>gnome.orca</li>
    <li>gnome.pomodoro</li>
    <li>gnomeExtensions.easyScreenCast</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.night-theme-switcher</li>
    <li>gobby (gobby5)</li>
    <li>goocanvasmm2</li>
    <li>gparted</li>
    <li>gsmartcontrol</li>
    <li>gtkmm4</li>
    <li>gtksourceviewmm4</li>
    <li>gtkspellmm</li>
    <li>guitarix</li>
    <li>horizon-eda</li>
    <li>inkscape</li>
    <li>inkscape-extensions.applytransforms</li>
    <li>inkscape-with-extensions</li>
    <li>iso-flags</li>
    <li>jack1</li>
    <li>jack2 (jack2Full)</li>
    <li>jackmix_jack1</li>
    <li>jstest-gtk</li>
    <li>kakounePlugins.rep</li>
    <li>koboredux</li>
    <li>koboredux-free</li>
    <li>kotatogram-desktop</li>
    <li>libgdamm</li>
    <li>littlegptracker</li>
    <li>lsp-plugins</li>
    <li>ltc-tools</li>
    <li>luksmeta</li>
    <li>luppp</li>
    <li>magnetophonDSP.CharacterCompressor</li>
    <li>magnetophonDSP.CompBus</li>
    <li>magnetophonDSP.ConstantDetuneChorus</li>
    <li>magnetophonDSP.LazyLimiter</li>
    <li>magnetophonDSP.MBdistortion</li>
    <li>magnetophonDSP.RhythmDelay</li>
    <li>magnetophonDSP.VoiceOfFaust</li>
    <li>magnetophonDSP.faustCompressors</li>
    <li>magnetophonDSP.pluginUtils</li>
    <li>magnetophonDSP.shelfMultiBand</li>
    <li>mapmap</li>
    <li>mate.mate-system-monitor</li>
    <li>mate.mate-utils</li>
    <li>mlt</li>
    <li>mojave-gtk-theme</li>
    <li>mooSpace</li>
    <li>nextcloud-client</li>
    <li>numix-cursor-theme</li>
    <li>numix-solarized-gtk-theme</li>
    <li>nwg-launchers</li>
    <li>obs-studio-plugins.obs-gstreamer</li>
    <li>octavePackages.audio</li>
    <li>open-vm-tools</li>
    <li>pantheon.elementary-session-settings</li>
    <li>paprefs</li>
    <li>pavucontrol</li>
    <li>pcb2gcode</li>
    <li>pdfchain</li>
    <li>pdfslicer</li>
    <li>performous</li>
    <li>phosh</li>
    <li>photoflow</li>
    <li>plata-theme</li>
    <li>pop-gtk-theme</li>
    <li>pulseeffects-legacy</li>
    <li>pulseview</li>
    <li>purple-mm-sms</li>
    <li>python38Packages.diagrams</li>
    <li>python38Packages.graph-tool</li>
    <li>python38Packages.inkex</li>
    <li>python38Packages.mlt</li>
    <li>python39Packages.diagrams</li>
    <li>python39Packages.graph-tool</li>
    <li>python39Packages.inkex</li>
    <li>python39Packages.mlt</li>
    <li>radiotray-ng</li>
    <li>rawtherapee</li>
    <li>shotcut</li>
    <li>solvespace</li>
    <li>sonic-pi</li>
    <li>soundtracker</li>
    <li>spring</li>
    <li>springLobby</li>
    <li>subtitleeditor</li>
    <li>tambura</li>
    <li>tang</li>
    <li>tdesktop</li>
    <li>tetraproc</li>
    <li>tuxguitar</li>
    <li>udiskie</li>
    <li>utsushi</li>
    <li>vimix-gtk-themes</li>
    <li>waybar</li>
    <li>wayfire</li>
    <li>wayfireApplications-unwrapped.wayfirePlugins.wf-shell</li>
    <li>wayfireApplications-unwrapped.wcm</li>
    <li>wcm</li>
    <li>webcamoid</li>
    <li>whatsapp-for-linux</li>
    <li>whitesur-gtk-theme</li>
    <li>workrave</li>
    <li>wpa_supplicant_gui</li>
  </ul>
</details>
<details>
  <summary>36 packages built:</summary>
  <ul>
    <li>ams-lv2</li>
    <li>ardour</li>
    <li>artyFX</li>
    <li>atkmm</li>
    <li>atkmm_2_36</li>
    <li>cairomm</li>
    <li>eq10q</li>
    <li>fme</li>
    <li>fmsynth</li>
    <li>ganv</li>
    <li>glibmm</li>
    <li>gnome2.gtkmm</li>
    <li>gnome2.libglademm</li>
    <li>gnome2.libgnomecanvasmm</li>
    <li>gnome2.pangomm_1_4</li>
    <li>gst_all_1.gstreamermm</li>
    <li>infamousPlugins</li>
    <li>ingen</li>
    <li>k3d</li>
    <li>libsigrok</li>
    <li>libsigrok_0_3</li>
    <li>libxmlxx</li>
    <li>libxmlxx3</li>
    <li>lightspark</li>
    <li>lv2-cpp-tools</li>
    <li>lvtk</li>
    <li>lxpanel</li>
    <li>nitrogen</li>
    <li>pangomm_2_48</li>
    <li>patchage</li>
    <li>rhvoice</li>
    <li>seq24</li>
    <li>sigrok-cli</li>
    <li>sorcer</li>
    <li>timekeeper</li>
    <li>vocproc</li>
  </ul>
</details>

**Comment by maxeaubrey** [comment]
Build failures seem unrelated and are fixed in #135414.

**Comment by maxeaubrey** [comment]
Result of `nixpkgs-review pr 136893` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>gnome.gnome-documents</li>
    <li>jucipp</li>
    <li>synfigstudio</li>
    <li>webbrowser</li>
    <li>xfce.xfce4-hardware-monitor-plugin</li>
    <li>xfce.xfce4_hardware_monitor_plugin</li>
  </ul>
</details>
<details>
  <summary>43 packages failed to build:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>ardour_5</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>astroid</li>
    <li>btrbk</li>
    <li>chrome-gnome-shell</li>
    <li>cinnamon.cinnamon-screensaver</li>
    <li>clevis</li>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>disorderfs</li>
    <li>emojione</li>
    <li>gigedit</li>
    <li>gnome.gnome-session (gnome.gnome_session)</li>
    <li>gnome.gnome-shell (gnome.gnome_shell)</li>
    <li>gnome.gnome-terminal (gnome.gnome_terminal)</li>
    <li>gnome.gnome-tweaks (gnome.gnome-tweak-tool)</li>
    <li>gnome.pomodoro</li>
    <li>gnomeExtensions.easyScreenCast</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.night-theme-switcher</li>
    <li>rep (kakounePlugins.rep)</li>
    <li>koboredux</li>
    <li>kotatogram-desktop</li>
    <li>luksmeta</li>
    <li>pantheon.elementary-session-settings</li>
    <li>pdfchain</li>
    <li>phosh</li>
    <li>photoflow</li>
    <li>python38Packages.diagrams</li>
    <li>python38Packages.graph-tool</li>
    <li>python39Packages.diagrams</li>
    <li>shotcut</li>
    <li>spring</li>
    <li>springLobby</li>
    <li>tang</li>
    <li>tdesktop</li>
    <li>tuxguitar</li>
    <li>udiskie</li>
    <li>vimix-gtk-themes</li>
    <li>whatsapp-for-linux</li>
    <li>whitesur-gtk-theme</li>
  </ul>
</details>
<details>
  <summary>148 packages built:</summary>
  <ul>
    <li>adriconf</li>
    <li>ahoviewer</li>
    <li>aj-snapshot</li>
    <li>ams-lv2</li>
    <li>arc-theme</li>
    <li>ardour</li>
    <li>artyFX</li>
    <li>arx-libertatis</li>
    <li>atkmm (atkmm_1_6 ,gnome2.atkmm_1_6)</li>
    <li>atkmm_2_36</li>
    <li>audiality2</li>
    <li>bibata-cursors</li>
    <li>bibata-cursors-translucent</li>
    <li>bibata-extra-cursors</li>
    <li>blueberry</li>
    <li>bombono</li>
    <li>cairomm (cairomm_1_0)</li>
    <li>capitaine-cursors</li>
    <li>carla</li>
    <li>cherrytree</li>
    <li>cinnamon.iso-flags-png-320x420</li>
    <li>cinnamon.iso-flags-svg</li>
    <li>dablin</li>
    <li>dblatexFull</li>
    <li>easyeffects</li>
    <li>eq10q</li>
    <li>faust2jack</li>
    <li>faust2jaqt</li>
    <li>faustPhysicalModeling</li>
    <li>faustStk</li>
    <li>ffado (libffado)</li>
    <li>fim</li>
    <li>flowblade</li>
    <li>fme</li>
    <li>fmsynth</li>
    <li>gImageReader</li>
    <li>ganv</li>
    <li>giada</li>
    <li>glibmm (glibmm_2_4 ,gnome2.glibmm_2_4)</li>
    <li>glom</li>
    <li>gnome.gnome-system-monitor</li>
    <li>gtkmm3 (gnome.gtkmm)</li>
    <li>gtksourceviewmm (gnome.gtksourceviewmm)</li>
    <li>nemiver (gnome.nemiver)</li>
    <li>orca (gnome.orca)</li>
    <li>gtkmm2 (gnome2.gtkmm)</li>
    <li>gnome2.libglademm</li>
    <li>gnome2.libgnomecanvasmm</li>
    <li>pangomm (gnome2.pangomm_1_4 ,pangomm_1_4)</li>
    <li>gobby</li>
    <li>goocanvasmm2</li>
    <li>gparted</li>
    <li>gsmartcontrol</li>
    <li>gst_all_1.gstreamermm</li>
    <li>gtkmm4</li>
    <li>gtksourceviewmm4</li>
    <li>gtkspellmm</li>
    <li>guitarix</li>
    <li>horizon-eda</li>
    <li>infamousPlugins</li>
    <li>ingen</li>
    <li>inkscape</li>
    <li>inkscape-extensions.applytransforms</li>
    <li>inkscape-with-extensions</li>
    <li>iso-flags</li>
    <li>jack1</li>
    <li>jack2</li>
    <li>jackmix_jack1</li>
    <li>jstest-gtk</li>
    <li>k3d</li>
    <li>koboredux-free</li>
    <li>libgdamm</li>
    <li>libsigrok</li>
    <li>libsigrok_0_3</li>
    <li>libxmlxx</li>
    <li>libxmlxx3</li>
    <li>lightspark</li>
    <li>littlegptracker</li>
    <li>lsp-plugins</li>
    <li>ltc-tools</li>
    <li>luppp</li>
    <li>lv2-cpp-tools</li>
    <li>lvtk</li>
    <li>lxpanel</li>
    <li>magnetophonDSP.CharacterCompressor</li>
    <li>magnetophonDSP.CompBus</li>
    <li>magnetophonDSP.ConstantDetuneChorus</li>
    <li>magnetophonDSP.LazyLimiter</li>
    <li>magnetophonDSP.MBdistortion</li>
    <li>magnetophonDSP.RhythmDelay</li>
    <li>magnetophonDSP.VoiceOfFaust</li>
    <li>magnetophonDSP.faustCompressors</li>
    <li>magnetophonDSP.pluginUtils</li>
    <li>magnetophonDSP.shelfMultiBand</li>
    <li>mapmap</li>
    <li>mate.mate-system-monitor</li>
    <li>mate.mate-utils</li>
    <li>mlt</li>
    <li>mojave-gtk-theme</li>
    <li>mooSpace</li>
    <li>nextcloud-client</li>
    <li>nitrogen</li>
    <li>numix-cursor-theme</li>
    <li>numix-solarized-gtk-theme</li>
    <li>nwg-launchers</li>
    <li>obs-studio-plugins.obs-gstreamer</li>
    <li>octavePackages.audio</li>
    <li>open-vm-tools</li>
    <li>pangomm_2_48</li>
    <li>paprefs</li>
    <li>patchage</li>
    <li>pavucontrol</li>
    <li>pcb2gcode</li>
    <li>pdfslicer</li>
    <li>performous</li>
    <li>plata-theme</li>
    <li>pop-gtk-theme</li>
    <li>pulseeffects-legacy</li>
    <li>pulseview</li>
    <li>purple-mm-sms</li>
    <li>python38Packages.inkex</li>
    <li>python38Packages.mlt</li>
    <li>python39Packages.graph-tool</li>
    <li>python39Packages.inkex</li>
    <li>python39Packages.mlt</li>
    <li>radiotray-ng</li>
    <li>rawtherapee</li>
    <li>rhvoice</li>
    <li>seq24</li>
    <li>sigrok-cli</li>
    <li>solvespace</li>
    <li>sonic-pi</li>
    <li>sorcer</li>
    <li>soundtracker</li>
    <li>subtitleeditor</li>
    <li>tambura</li>
    <li>tetraproc</li>
    <li>timekeeper</li>
    <li>utsushi</li>
    <li>vocproc</li>
    <li>waybar</li>
    <li>wayfire</li>
    <li>wayfirePlugins.wf-shell (wayfireApplications-unwrapped.wayfirePlugins.wf-shell)</li>
    <li>wayfireApplications-unwrapped.wcm</li>
    <li>wcm</li>
    <li>webcamoid</li>
    <li>workrave</li>
    <li>wpa_supplicant_gui</li>
  </ul>
</details>

**Comment by maxeaubrey** [comment]
I seem to be getting a lot of transient failures in python packages (watchdog, fs...), mostly in tests - which isn't even related to the PR afaict.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -15983,14 +15983,16 @@ with pkgs;
 
   atk = callPackage ../development/libraries/atk { };
 
-  atkmm = callPackage ../development/libraries/atkmm { };
+  atkmm = atkmm_1_6;
+  atkmm_1_6 = callPackage ../development/libraries/atkmm/1.6.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:15987]:
```suggestion
  atkmm = callPackage ../development/libraries/atkmm/1.6.nix { };
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -15965,8 +15965,8 @@ with pkgs;
     dbus = dbus.override { systemd = null; };
   });
 
-  glibmm = callPackage ../development/libraries/glibmm { };
-
+  glibmm = glibmm_2_4;
+  glibmm_2_4 = callPackage ../development/libraries/glibmm/2.4.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:15969]:
```suggestion
  glibmm = callPackage ../development/libraries/glibmm/2.4.nix { };
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -15983,14 +15983,16 @@ with pkgs;
...
+  atkmm_2_36 = callPackage ../development/libraries/atkmm/2.36.nix { };
 
   pixman = callPackage ../development/libraries/pixman { };
 
   cairo = callPackage ../development/libraries/cairo { };
 
-  cairomm = callPackage ../development/libraries/cairomm { };
-
+  cairomm = cairomm_1_0;
+  cairomm_1_0 = callPackage ../development/libraries/cairomm/1.0.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:15995]:
```suggestion
  cairomm = callPackage ../development/libraries/cairomm/1.0.nix { };
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -16003,10 +16005,10 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) Carbon Cocoa;
   };
 
-  pangomm = callPackage ../development/libraries/pangomm {
+  pangomm = pangomm_1_4;
+  pangomm_1_4 = callPackage ../development/libraries/pangomm/1.4.nix {
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:16009]:
```suggestion
  pangomm = callPackage ../development/libraries/pangomm/1.4.nix {
```

File: pkgs/development/libraries/atkmm/1.6.nix [hunk 5]
```diff
@@ -1,19 +1,19 @@
-{ lib, stdenv, fetchurl, atk, glibmm, pkg-config, gnome }:
+{ lib, stdenv, fetchurl, atk, glibmm_2_4, pkg-config, gnome, meson, ninja, python3 }:
 
 stdenv.mkDerivation rec {
   pname = "atkmm";
-  version = "2.28.0";
+  version = "2.28.2";
```
[the remark this answers is not in the document]
Comment (maxeaubrey) [reply pkgs/development/libraries/atkmm/1.6.nix:5]: 1.6 refers to the ABI version.

File: pkgs/development/libraries/atkmm/2.36.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  passthru = {
+    updateScript = gnome.updateScript {
+      packageName = pname;
+      versionPolicy = "odd-unstable";
+    };
+  };
+
+  meta = {
+    description = "C++ wrappers for ATK accessibility toolkit";
+    license = lib.licenses.lgpl21Plus;
```
[the remark this answers is not in the document]
Comment (maxeaubrey) [reply pkgs/development/libraries/atkmm/2.36.nix:29]: Same as `1.6` then :)

**Comment by maxeaubrey** [comment]
@SuperSandro2000 Are you suggesting I drop the rename commits? Half the point of this PR was making a clearer distinction between ABI versions.

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 Are you suggesting I drop the rename commits? Half the point of this PR was making a clearer distinction between ABI versions.

Yeah, I think it is enough to have the latest version and some older version for packages which do not work with the new version. Ideally we would want to get rid of the older versions again and not keep it forever. Also it is unlikely that we have multiple other versions in the future that keep changing.

**Comment by maxeaubrey** [comment]
We still have gtkmm2, it's unlikely these older versions will be deprecated all that quickly.

File: pkgs/desktops/gnome-2/default.nix [hunk 7]
```diff
@@ -71,7 +71,7 @@ lib.makeScope pkgs.newScope (self: with self; {
 } // lib.optionalAttrs (config.allowAliases or true) {
   inherit (pkgs)
     # GTK Libs
-    glib glibmm atk atkmm cairo pango pangomm gdk_pixbuf gtkmm2 libcanberra-gtk2
+    glib glibmm_2_4 atk atkmm_1_6 cairo pango pangomm_1_4 gdk_pixbuf gtkmm2 libcanberra-gtk2
```
Comment (jtojnar) [inline pkgs/desktops/gnome-2/default.nix:74]: No point in legacy aliases since nobody would use `gnome2.glibmm_2_4`.

**Pull request closed without merging**